tests/integration/utils2.go

```diff
@@ -342,7 +341,7 @@ func performAction(
 		assertClientIntrospectionResults(s, action)
 
 	case WaitForSync:
-		waitForSync(s, action)
+		waitForMergeEvents(s, action)
```

Comment:
praise: it's great that now waiting is done on the same thread. Earlier it used to call a goroutine for every `ConnectPeers` action and had subtle shortcomings.